### ISSUE TITLE
Using gettimeofday() in PortTransmitter::udelay() function

### ISSUE THREAD

**Issue by ksv141** [pstavirs/ostinato#211]
In such application it's more preferable to use clock_gettime() instead of gettimeofday() because

> The time returned by gettimeofday() is affected by discontinuous jumps in the system time (e.g., if the system administrator manually changes the system time).  If you need a monotonically increasing      clock, see clock_gettime(2).

see http://man7.org/linux/man-pages/man2/gettimeofday.2.html

This can be potentially the bug

**Comment by pstavirs** [comment]
Yes, the bug is possible - if the change happens while transmit is running on a port.

However, there are other considerations when choosing a timing function -
* Availability and support across *nix platforms - Linux, Unix, OSX, BSD
* Time taken by the timing function itself i.e. the overhead cost of a timing call

If you can provide evidence that clock_gettime() or some other API is better suited on the above considerations, I will consider making the change.

**Comment by ksv141** [comment]
for gettimeofday it is not guaranteed an usec precision becuase It's just wall-clock for general purpose.
POSIX [recommend](http://pubs.opengroup.org/onlinepubs/9699919799/functions/gettimeofday.html) to use the clock_gettime() function instead of the gettimeofday() and marks this function as obsolete.
